packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/describers.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { LocalComponent } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { join } from 'path';
+import { RetrieveDescriber } from '.';
+import { BrowserNode } from '../../orgBrowser';
+import { SfdxPackageDirectories } from '../../sfdxProject';
+
+abstract class NodeDescriber implements RetrieveDescriber {
+  protected node: BrowserNode;
+
+  constructor(node: BrowserNode) {
+    this.node = node;
+  }
+
+  public abstract buildMetadataArg(): string;
+
+  public abstract gatherOutputLocations(): Promise<LocalComponent[]>;
+
+  protected async buildOutput(node: BrowserNode): Promise<LocalComponent[]> {
+    const typeNode = node.getAssociatedTypeNode();
+    // create cmp for every package directory to accommodate for the cli retrieve bug
+    const packageDirectories = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    return packageDirectories.map(directory => ({
+      fileName: node.fullName,
+      outputdir: join(directory, 'main', 'default', typeNode.directoryName!),
+      type: typeNode.fullName,
+      suffix: typeNode.suffix
+    }));
+  }
+}
+
+class TypeNodeDescriber extends NodeDescriber {
+  public buildMetadataArg(data?: LocalComponent[]): string {
+    if (data) {
+      const dedupe = new Set<string>(); // filter dupes caused by cli bug. See buildOutput in parent class
+      data.forEach(c => dedupe.add(`${c.type}:${c.fileName}`));
```

Comment:
Because there can be multiple components that actually represent one (just with different package directories), we need to dedupe them. Once the CLI bug is fixed we can remove this